api/v1alpha5/virtualmachinesnapshot_types.go

```diff
@@ -139,6 +139,31 @@ type VirtualMachineSnapshotStorageStatus struct {
 	Requested []VirtualMachineSnapshotStorageStatusRequested `json:"requested,omitempty"`
 }
 
+type VirtualMachineSnapshotReferenceType string
+
+const (
+	// VirtualMachineSnapshotReferenceTypeManaged represents a snapshot that is managed by the Supervisor Cluster.
+	VirtualMachineSnapshotReferenceTypeManaged VirtualMachineSnapshotReferenceType = "Managed"
+
+	// VirtualMachineSnapshotReferenceTypeUnmanaged represents a snapshot that is managed by the backing infrastructure (e.g., vSphere).
+	VirtualMachineSnapshotReferenceTypeUnmanaged VirtualMachineSnapshotReferenceType = "Unmanaged"
```

Comment:
Is this not possible to test via vcsim? I don't see this being used in any of the tests

Reply:
It's not tested with vcsim, but tested with mocked moVM in update_status_test.go
https://github.com/vmware-tanzu/vm-operator/pull/1133/files#diff-fab70a3f39d63d3c216b52ec02f18f299ef5767ffcb982d22d998963a73b4af3R3966